src/ucp/wireup/select.c

```diff
@@ -1039,30 +1046,74 @@ ucp_wireup_add_amo_lanes(const ucp_wireup_select_params_t *select_params,
                                           UCP_LANE_TYPE_AMO, select_ctx);
 }
 
-static double ucp_wireup_am_score_func(const ucp_worker_iface_t *wiface,
+static double ucp_wireup_am_score_func(const ucp_wireup_criteria_t *self,
+                                       const ucp_worker_iface_t *wiface,
                                        const uct_md_attr_t *md_attr,
-                                       const ucp_address_iface_attr_t *remote_iface_attr)
+                                       const ucp_address_entry_t *remote_addr)
 {
     /* best end-to-end latency */
     return 1e-3 / (ucp_wireup_tl_iface_latency(wiface->worker->context,
                                                &wiface->attr,
-                                               remote_iface_attr) +
-                   wiface->attr.overhead + remote_iface_attr->overhead);
+                                               &remote_addr->iface_attr) +
+                   wiface->attr.overhead + remote_addr->iface_attr.overhead);
+}
+
+static double ucp_tl_iface_bandwidth_ratio(ucp_context_h context,
+                                           unsigned dev_count,
+                                           unsigned num_paths)
+{
+    double ratio = context->config.ext.multi_path_ratio * dev_count;
+
+    if (UCS_CONFIG_DBL_IS_AUTO(context->config.ext.multi_path_ratio)) {
+        ratio = (double)dev_count / (double)num_paths;
+    } else {
+        ratio = context->config.ext.multi_path_ratio * dev_count;
+    }
+
+    return ucs_max(0.1, 1.0 - ratio);
```

Comment:
why 0.1 and not 0 ?

Reply:
it makes divider to be 0 for score calculations. I'd use some small value, indicating that some bw does exist for extra connections

Reply:
maybe convert to multiplier so score would be 0 in this case?
if cant set something like 1e-5

Reply:
ok